cartography/intel/kubernetes/services.py

```diff
@@ -0,0 +1,85 @@
+            "name": service.metadata.name,
+            "creation_timestamp": get_epoch(service.metadata.creation_timestamp),
+            "deletion_timestamp": get_epoch(service.metadata.deletion_timestamp),
+            "namespace": service.metadata.namespace,
+            "cluster_uid": cluster["uid"],
+            "type": service.spec.type,
+            "selector": service.spec.selector,
+            "load_balancer_ip": service.spec.load_balancer_ip,
+        }
+
+        ingresses = service.status.load_balancer.ingress
+        for ingress in ingresses or list():
+            item.update({"ingress_host": ingress.hostname, "ingress_ip": ingress.ip})
+
+        service_pods = list()
+        for pod in pods:
+            is_service_pod = True if service.spec.selector else False
+            for selector in service.spec.selector or dict():
+                if (
+                    selector not in pod["labels"] or
+                    service.spec.selector[selector] != pod["labels"][selector]
+                ):
+                    is_service_pod = False
+                    break
+            if is_service_pod:
+                service_pods.append(pod)
+        item["pods"] = service_pods
+        services.append(item)
+    load_service_data(session, services, update_tag)
+
+
+def load_service_data(session: Session, data: List[Dict], update_tag: int) -> None:
+    ingestion_cypher_query = """
+    UNWIND {services} as k8service
+        MERGE (service:KubernetesService {id: k8service.uid})
+        ON CREATE SET service.firstseen = timestamp()
+        SET service.lastupdated = {update_tag},
+            service.name = k8service.name,
+            service.created_at = k8service.creation_timestamp,
+            service.deleted_at = k8service.deletion_timestamp,
+            service.type = k8service.type,
+            service.load_balancer_ip = k8service.load_balancer_ip,
+            service.ingress_host = k8service.ingress_host,
+            service.ingress_ip = k8service.ingress_ip
+        WITH service, k8service.namespace as ns, k8service.cluster_uid as cuid, k8service.pods as k8pods
+        MATCH (space:KubernetesNamespace {name: ns})-[:IN_CLUSTER]->(cluster:KubernetesCluster {id: cuid})
+        MERGE (space)-[rel1:HAS_SERVICE]->(service)
+        ON CREATE SET rel1.firstseen = timestamp()
+        SET rel1.lastupdated = {update_tag}
+        MERGE (service)-[rel2:IN_NAMESPACE]->(space)
+        ON CREATE SET rel2.firstseen = timestamp()
+        SET rel2.lastupdated = {update_tag}
+        WITH service, k8pods
+        UNWIND k8pods as k8pod
+            MATCH (pod:KubernetesPod {id: k8pod.uid})
+            MERGE (service)-[rel3:SERVES_POD]->(pod)
+            ON CREATE SET rel3.firstseen = timestamp()
+            SET rel3.lastupdated = {update_tag}
+            MERGE (pod)-[rel4:BELONGS_TO]->(service)
+            ON CREATE SET rel4.firstseen = timestamp()
+            SET rel4.lastupdated = {update_tag}
```

Comment:
No need to have two way relationships. If a Service Serves a Pod, it is implicit a pod belongs to a service. Just make one relationship here.